src/util/ConfigManager/ConfigOption.cpp

```diff
@@ -0,0 +1,314 @@
+      [&json, this]<typename T>(const Data<T>&) { setValue(json.get<T>()); },
+      data_);
+  configurationOptionWasSet_ = true;
+}
+
+// ____________________________________________________________________________
+std::string_view ConfigOption::getIdentifier() const {
+  return static_cast<std::string_view>(identifier_);
+}
+
+// ____________________________________________________________________________
+std::string ConfigOption::contentOfAvailableTypesToString(
+    const std::optional<AvailableTypes>& value) {
+  if (value.has_value()) {
+    // Converts a `AvailableTypes` to their string representation.
+    auto availableTypesToString = []<typename T>(
+                                      const T& content,
+                                      auto&& variantSubTypetoString) {
+      // Return the internal value of the `std::optional`.
+      if constexpr (std::is_same_v<T, std::string>) {
+        // Add "", so that it's more obvious, that it's a string.
+        return absl::StrCat("\"", content, "\"");
+      } else if constexpr (std::is_same_v<T, bool>) {
+        return content ? std::string{"true"} : std::string{"false"};
+      } else if constexpr (std::is_arithmetic_v<T>) {
+        return std::to_string(content);
+      } else {
+        // Is must be a vector.
+        static_assert(ad_utility::isVector<T>);
+
+        /*
+        We have to use the EXPLICIT type, because a vector of bools uses a
+        special 1 bit data type for it's entry references, instead of a bool. So
+        an `auto` would land the recursive call right back in this else case.
+        */
+        using VectorEntryType = T::value_type;
+
+        std::ostringstream stream;
+        stream << "{";
+        forEachExcludingTheLastOne(
+            content,
+            [&stream, &variantSubTypetoString](const VectorEntryType& entry) {
+              stream << variantSubTypetoString(entry, variantSubTypetoString)
+                     << ", ";
+            },
+            [&stream, &variantSubTypetoString](const VectorEntryType& entry) {
+              stream << variantSubTypetoString(entry, variantSubTypetoString);
+            });
+        stream << "}";
+        return stream.str();
+      }
+    };
+
+    return std::visit(
+        [&availableTypesToString](const auto& v) {
+          return availableTypesToString(v, availableTypesToString);
+        },
+        value.value());
+  } else {
+    return "None";
+  }
+}
+// ____________________________________________________________________________
+std::string ConfigOption::getValueAsString() const {
+  return std::visit(
+      [](const auto& d) {
+        return contentOfAvailableTypesToString(*d.variablePointer_);
+      },
+      data_);
+}
+
+// ____________________________________________________________________________
+nlohmann::json ConfigOption::getValueAsJson() const {
+  return std::visit(
+      [](const auto& d) { return nlohmann::json(*d.variablePointer_); }, data_);
+}
+
```

Comment:
These functions should check whether the `variablePointter` was previously set.

Reply:
Made the changes, see [c9bc474](https://github.com/ad-freiburg/qlever/pull/977/commits/c9bc474cd93a5f23530d3b480c96bd5cea42d16a).